engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/helper/MockProvider.java

```diff
@@ -3246,13 +3246,13 @@ public static List<BatchStatistics> createMockBatchStatisticsList() {
   public static MessageCorrelationResult createMessageCorrelationResult(MessageCorrelationResultType type) {
     MessageCorrelationResult result = mock(MessageCorrelationResult.class);
     when(result.getResultType()).thenReturn(type);
-    if (result.getResultType().equals(MessageCorrelationResultType.Execution)) {
-      Execution ex = createMockExecution();
-      when(result.getExecution()).thenReturn(ex);
-    } else {
-      ProcessInstance instance = createMockInstance();
-      when(result.getProcessInstance()).thenReturn(instance);
-    }
+
+    Execution ex = createMockExecution();
+    when(result.getExecution()).thenReturn(ex);
+
+    ProcessInstance instance = createMockInstance();
+    when(result.getProcessInstance()).thenReturn(instance);
+
```

Comment:
👍 for adjusting the mock for the `MessageCorrelationResult`.
❌ The same adjustment is necessary for the `MessageCorrelationResultWithVariables` in the method below.

Reply:
Hi [Yana](@yanavasileva ),
Thank you for catching the changes required for the test of MessageCorrelationResultWithVariables. I've made the suggested changes.